File: old_contents/boj_codes/Python3/18222.py
# https://www.acmicpc.net/problem/18222
# 20201122

import sys
import math

# N -= 2^(floor(lgN))
def find_complement(N):
    floor_log = math.ceil(math.log(N, 2)) - 1
    closest_point = 2 ** floor_log

    return N - closest_point


if __name__ == "__main__":
    N = int(sys.stdin.readline().rstrip())
    is_reversed = False

    while N != 1:
        N = find_complement(N)
        is_reversed = not is_reversed

    print(1) if is_reversed else print(0)

File: old_contents/boj_codes/C++17/18222.cpp
// https://www.acmicpc.net/problem/18222
// First Written : 20210129
// Last Modified : 20210129

#include <cstdio>
#include <cmath>

long long find_complement(long long idx){
    long long floor_log = (long long)ceil(log2((double)idx));
    long long closest = (long long)pow(2, floor_log - 1);

    return idx - closest;
}

int main(){
    long long N;
    scanf("%lld", &N);
    bool is_reversed = false;

    while(N != 1){
        N = find_complement(N);
        is_reversed = !is_reversed;
    }

    if(is_reversed)
        printf("1");
    else
        printf("0");
    return 0;
}